management/backup.py

```diff
@@ -456,6 +456,23 @@ def list_target_files(config):
 			raise ValueError(e.reason)
 
 		return [(key.name[len(path):], key.size) for key in bucket.list(prefix=path)]
+	elif target.scheme == 'b2':
```

Comment:
Have you tried B2's new S3-compatible API? I'm not 100% sure whether it will work here as it's still a beta and missing some features but it might be worth a shot: https://www.backblaze.com/blog/backblaze-b2-s3-compatible-api/

Reply:
Yes, I tried it out. I decided against it for two reasons:

* I think it's easier for the user to enter his/her backblaze credentials instead of fiddling around with the s3-compatible API. If you are not familiar with the s3 compatible API I don't think it's obvious to use.

* I couldn't get the s3 compatible API working with the duplicity version shipped with ubuntu. I think it does not work out of the box and one would to adjust miab anyways. If you just do the differentiation between S3/B2 in the frontend, then you need to do pre- and postprocessing in the background and I don't think that it helps keeping the code clean.

But that's just my point of view and I am happy to hear counter-arguments :)

Reply:
Thanks for the clarification. I entirely agree